src/bp/core/services/nlu/intent-service.ts

```diff
@@ -42,15 +49,15 @@ export class IntentService {
         const conditions = tn?.conditions.filter(x => x?.id === 'user_intent_is')
 
         for (let i = 0; i < conditions.length; i++) {
-          const intentName = sanitizeFileName(`${flow.name}/${tn?.name}/${i}`) // TODO: remove this, move name generation to front-end
+          const intentName = sanitizeFileName(`${topicName}/${flow.name}/${tn?.name}/${i}`)
           if (intentsByName[intentName]) {
             throw new Error(`Duplicated intent with name "${intentName}"`)
           }
           intentsByName[intentName] = {
             contexts: [topicName],
             filename: flow.name,
             name: intentName,
-            slots: flow.variables?.map(x => ({ name: x.params?.name, entity: x?.params?.subType ?? x.type })) ?? [],
+            slots: flow.variables?.map(x => ({ name: x.params?.name, entity: x.type })) ?? [], // TODO: sub-type (type of list entity and type of pattern is not included in variables yet)
```

Comment:
Here `x.type` will be "enum" and you will not have the correct name of the entity. It really is subType